src/profiler.cpp

```diff
@@ -149,8 +149,7 @@ void Profiler::checkDeadline() {
         ssize_t w = write(STDERR_FILENO, error, sizeof(error) - 1);
         (void) w;
 
-        _running = false;
-        setTimer(0, 0);
+        stop();
```

Comment:
It is not safe to call Profiler::stop() from within signal handler.
Automatic stop mechanism should be either re-implemented  (e.g. by sleeping in a separate thread) or removed altogether. It's OK to remove it for now and fix later in a separate issue.